api/openapi-spec/v3/api__v1_openapi.json

```diff
@@ -26610,7 +26673,8 @@
                 "$ref": "#/components/schemas/io.k8s.apimachinery.pkg.apis.meta.v1.Patch"
               }
             }
-          }
+          },
+          "required": true
```

Comment:
what is this used for? the schema for patch doesn't look right for apply (can be yaml, not required to be json) or `application/json-patch+json` (is a list, not an object)

Reply:
communicating to the user that a requestbody is required for patch operations, this is something that we have in the openapi v2 schema. It's currently not used for anything other than for informational purposes.

As for the incorrect schema, cc @apelisse
Is the list here incorrect? https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/types/patch.go#L28 
I see we only accept one format (json or yaml) for each of the patches, do we need to expand this list?

Reply:
We don't take json for apply, so that's correct, but the schema type is wrong indeed, it should be the type of the object (same as returned type). I think that's already true about v2 though. I'll open an issue for that.

Reply:
After looking into the code a bit more, the idea of consuming a "Patch" object instead of the return object schema seems intentional.
Per https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1200-L1201

`// Patch is provided to give a concrete name and type to the Kubernetes PATCH request body.`

Due to limitations with go-restful, we're not able to specify different return types for different patch types. I'm testing out a PR to change the patch return type, though not sure if the OpenAPI is more accurate.

Reply:
Yeah, I don't disagree.

Reply:
I didn't realize openapi v2 was already publishing body parameters with `required: true`

if that's been the case for multiple releases already, bringing this to parity seems ok. I checked and v2 has indicated body param is required (including the patch one) since at least 1.25